test/functional/tests/seq_cutoff/test_seq_cutoff.py

```diff
@@ -0,0 +1,253 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+
+import pytest
+import random
+from enum import Enum, auto
+from api.cas import casadm
+from api.cas.core import SEQ_CUTOFF_THRESHOLD_MAX
+from api.cas.cache_config import SeqCutOffPolicy, CacheMode
+from api.cas.casadm import StatsFilter
+from core.test_run import TestRun
+
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Size, Unit
+from test_utils.os_utils import Udev, sync
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+
+
+class VerifyType(Enum):
+    NEGATIVE = auto()
+    POSITIVE = auto()
+    EQUAL = auto()
+
+
+@pytest.mark.parametrize("cache_mode, io_type, io_type_last", [
+    (CacheMode.WB, ReadWrite.write, ReadWrite.randwrite),
+    (CacheMode.WT, ReadWrite.write, ReadWrite.randwrite),
+    (CacheMode.WA, ReadWrite.read, ReadWrite.randread),
+    (CacheMode.WO, ReadWrite.write, ReadWrite.randwrite)])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_seq_cutoff_multi_core(cache_mode, io_type, io_type_last):
+    """
+    title: Sequential cut-off tests during sequential and random IO 'always' policy with 4 cores
+    description: |
+        Testing if amount of data written to cache after sequential writes for different
+        sequential cut-off thresholds on each core, while running sequential IO on 3 out of 4
+        cores and random IO against the last core, is correct.
+    pass_criteria:
+        - Amount of written blocks to cache is less or equal than amount set
+          with sequential cut-off threshold for three first cores.
+        - Amount of written blocks to cache is equal sequential to io size run against last core.
+    """
+    with TestRun.step("Test prepare (start cache and add cores)"):
+        cache, cores = prepare(cores_count=4)
+        writes_before = []
+        io_sizes = []
+        thresholds = []
+        fio_additional_size = Size(10, Unit.Blocks4096)
+        for i in range(len(cores)):
+            thresholds.append(Size(random.randint(1, int((cache.cache_device.size
+                                                          * 0.5).get_value(Unit.KibiByte)))))
```

Comment:
Please don't generate random parameters inside test case. This creates non-reproducible test cases. Use pytest parametrize instead (even if it's just oneshot per test case).